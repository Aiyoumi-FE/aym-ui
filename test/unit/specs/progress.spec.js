import Vue from 'vue'
import { mount, shallow } from '@vue/test-utils'

import Progress from '@/components/progress/progress'

describe('Progress', () => {
    describe('step', () => {
        let cmp

        beforeEach(() => {
            cmp = mount(Progress)
        })
        afterEach(() => {
            cmp.destroy()
        })
        it('type:step', () => {
            expect(cmp.classes()).toContain('mui-progress_step')

        })
        // cmp.setProps({ type: 'step' })
        it('stepText：显示文本', () => {
            expect(cmp.text()).toContain('012')
            cmp.setProps({ stepText: ['a1', 'a2', 'a3'] })
            expect(cmp.text()).toContain('a1a2a3')

        })
        it('curStep：当前所在步数', () => {
            cmp.setProps({ stepText: ['b1', 'b2', 'b3', 'b4', 'b5', 'b6'] })
            const AllLi = cmp.findAll('li')
            expect(AllLi.at(0).classes()).toContain('cur')

            const activeLi = (len) => AllLi.filter((item, index) => {
                if (index <= len) {
                    // 确保是前几个
                    expect(AllLi.at(index).classes()).toContain('cur')
                }
                return item.classes().includes('cur')
            })
            for (let i = 1; i < 6; i++) {
                cmp.setProps({ curStep: i })
                expect(activeLi(i).length).toBe(i + 1)
            }
            expect(cmp.vm.$el).toMatchSnapshot('满格')
        })
    })
    describe('timeline', () => {
        let cmp
        const timelineArr = [{
            date: '3.28',
            title: '第1步标题',
            desc: '第1步描述'
        }, {
            date: '3.29',
            title: '第2步标题',
            desc: '第2步描述'
        }]
        beforeEach(() => {
            cmp = mount(Progress, {
                propsData: {
                    type: 'timeline',
                    stepText: timelineArr
                }
            })
        })
        afterEach(() => {
            cmp.destroy()
        })
        it('type:timeline', () => {
            expect(cmp.classes()).toContain('mui-progress_timeline')

        })

        it('默认不显示一半线', () => {

            const AllLi = cmp.findAll('li')
            expect(AllLi.at(0).classes()).toContain('none-halfline')
            cmp.setProps({ // ???? curStep 和type 的位置先后居然会影响循环渲染？？？？
                // curStep 在下的话，样式会在下一轮作渲染（nextTick）,抽空查一下test-util, 和初始化prop也有关系
                curStep: 1

            })
            expect(AllLi.at(1).classes()).toContain('none-halfline')
            expect(cmp.vm.$el).toMatchSnapshot('不显示那半条线')

        })
        it('显示那半条线', () => {
            cmp.setProps({
                isHalfLine: false
            })
            // return cmp.vm.$nextTick(() => { // ???????
            expect(cmp.findAll('li').at(0).classes()).not.toContain('none-halfline')
            expect(cmp.vm.$el).toMatchSnapshot('显示那半条线')

            //  })
        })
    })
})
