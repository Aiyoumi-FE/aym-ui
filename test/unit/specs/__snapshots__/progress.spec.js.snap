// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress step curStep：当前所在步数: 满格 1`] = `
<ul
  class="mui-progress_step"
>
  <li
    class="mui-progress__item cur"
  >
    <span
      class="mui-progress__ball"
    >
      <span
        class="mui-progress__txt"
      >
        b1
      </span>
    </span>
  </li>
  <li
    class="mui-progress__item cur"
  >
    <span
      class="mui-progress__ball"
    >
      <span
        class="mui-progress__txt"
      >
        b2
      </span>
    </span>
  </li>
  <li
    class="mui-progress__item cur"
  >
    <span
      class="mui-progress__ball"
    >
      <span
        class="mui-progress__txt"
      >
        b3
      </span>
    </span>
  </li>
  <li
    class="mui-progress__item cur"
  >
    <span
      class="mui-progress__ball"
    >
      <span
        class="mui-progress__txt"
      >
        b4
      </span>
    </span>
  </li>
  <li
    class="mui-progress__item cur"
  >
    <span
      class="mui-progress__ball"
    >
      <span
        class="mui-progress__txt"
      >
        b5
      </span>
    </span>
  </li>
  <li
    class="mui-progress__item cur"
  >
    <span
      class="mui-progress__ball"
    >
      <span
        class="mui-progress__txt"
      >
        b6
      </span>
    </span>
  </li>
</ul>
`;

exports[`Progress timeline 显示那半条线: 显示那半条线 1`] = `
<ul
  class="mui-progress_timeline"
>
  <li
    class="mui-progress__item cur"
  >
    <div
      class="mui-progress__time"
    >
      <span>
        3.28
      </span>
       
      <span
        class="mui-progress__ball"
      />
    </div>
     
    <div
      class="mui-progress__content"
    >
      <p>
        第1步标题
      </p>
       
      <p>
        第1步描述
      </p>
    </div>
  </li>
  <li
    class="mui-progress__item"
  >
    <div
      class="mui-progress__time"
    >
      <span>
        3.29
      </span>
       
      <span
        class="mui-progress__ball"
      />
    </div>
     
    <div
      class="mui-progress__content"
    >
      <p>
        第2步标题
      </p>
       
      <p>
        第2步描述
      </p>
    </div>
  </li>
</ul>
`;

exports[`Progress timeline 默认不显示一半线: 不显示那半条线 1`] = `
<ul
  class="mui-progress_timeline"
>
  <li
    class="mui-progress__item cur"
  >
    <div
      class="mui-progress__time"
    >
      <span>
        3.28
      </span>
       
      <span
        class="mui-progress__ball"
      />
    </div>
     
    <div
      class="mui-progress__content"
    >
      <p>
        第1步标题
      </p>
       
      <p>
        第1步描述
      </p>
    </div>
  </li>
  <li
    class="mui-progress__item cur none-halfline"
  >
    <div
      class="mui-progress__time"
    >
      <span>
        3.29
      </span>
       
      <span
        class="mui-progress__ball"
      />
    </div>
     
    <div
      class="mui-progress__content"
    >
      <p>
        第2步标题
      </p>
       
      <p>
        第2步描述
      </p>
    </div>
  </li>
</ul>
`;
